Manuals/Altibase_7.1/kor/Replication Manual.md

```diff
@@ -2042,28 +2042,14 @@ ALTER REPLICATION replication_name START WITH OFFLINE;
 
 ##### 설명
 
-Active-Standby 이중화 환경에서, 서비스를 제공하는 Active 서버에서 장애가
-발생하면 Standby 서버에 로그를 전송할 수 없게 된다. 이 때 오프라인 옵션을
-사용하여 미전송된 로그를 Standby 서버에 반영할 수 있다. Active 서버에 장애가
-발생하면 즉, Active 서버가 Standby 서버로 로그를 전송할 수 없는 상황이면,
-Standby 서버가 오프라인 옵션으로 Active 서버의 로그 파일에 직접 접근하여 미전송
-로그를 가져와서 반영할 수 있다. 단, Active 서버가 Standby 서버 쪽으로 한번이라도
-이중화를 수행한 적이 있는 경우에만, Standby 서버에서 오프라인 옵션을 사용할 수
-있다.
+Active-Standby 이중화 환경에서, 서비스를 제공하는 Active 서버에서 장애 발생으로 Altibase 서버가 중지되면 Active 서버의 송신자는 XLog 변환 및 전송을 ~~Standby 서버에 로그를 전송할 수 없게 된다~~할 수 없다. 이 때 Standby 서버에서 오프라인 옵션을 사용하여 ~~미전송된 로그를~~ 반영하지 못한 트랜잭션을 Standby 서버에 반영할 수 있다.  Standby 서버에서 이중화 객체에 오프라인 옵션을 설정하고 이중화를 시작하면 오프라인 송신자가 시작된다. 오프라인 송신자는 Active 서버의 송신자 역할을 대신하며 이를 오프라인 이중화라고 한다.  ~~Active 서버에 장애가 발생하면 즉, Active 서버가 Standby 서버로 로그를 전송할 수 없는 상황이면, Standby 서버가 오프라인 옵션으로 Active 서버의 로그 파일에 직접 접근하여 미전송 로그를 가져와서 반영할 수 있다.~~ 단, Active 서버가 Standby 서버 쪽으로 한번이라도 이중화를 수행한 적이 있는 경우에만, Standby 서버에서 오프라인 옵션을 사용할 수 있다.
```

Comment:
단, Active 서버가 Standby 서버 쪽으로 한번이라도 이중화를 수행한 적이 있는 경우에만, Standby 서버에서 오프라인 옵션을 사용할 수 있다.
--> 단, Active 서버가 Standby 서버 쪽으로 **구동 후에** 한번이라도 이중화를 수행한 적이 있는 경우에만, Standby 서버에서 오프라인 옵션을 사용할 수 있다.
위의 내용이 기술적으로 Standby 서버 재시작 마다 해당되는 내용입니다. 
자세히 설명드리면, Standby 서버의 오프라인 이중화 재시작 지점과 다른 부가 정보들은 메모리에 유지되는데, 
Active에서 송신자가 Standby로 접속 할때 생성됩니다. 그리고, 재구동 하면 사라지므로, 위의 설명에서 "구동후에" 라는 멘트를 넣어야 할 것으로 보입니다.

Reply:
> 

이중화 객체 생성하고 시작하고 변경 트랜잭션이 발생하지 않는 상태에서 ALTER REPLICATION rep START WITH OFFLINE; 했는데, Alter success. 라고 나옵니다. 

위 문장은 이 경우는 해당하지 않는 것인가요? 
위 문장은 주의 사항에 해당하는 것인가요? 위 문장대로 하지 않았다고해서 오프라인 이중화 시작이 실패하거나 하지는 않는 것 같아서 제약 사항은 아닌 듯 한데... 
확인 부탁드립니다.

Reply:
"단, Active 서버가 Standby 서버 쪽으로 한 번이라도 이중화를 수행한 적이 있는 경우에만 Standby 서버에서 오프라인 옵션을 사용할 수 있다." 

이 문장은 내용 검토 후 주의 사항으로 이동할까 합니다.

Reply:
이중화 객체 생성하고 시작하고 변경 트랜잭션이 발생하지 않는 상태에서 ALTER REPLICATION rep START WITH OFFLINE; 했는데, Alter success. 라고 나옵니다. ==> 이건 정상 상황이며, 제가 설명한 상황에 해당하지 않습니다. 
정확히 다시 말씀드리면, 
- 이중화 객체 생성하고 시작하고 변경 트랜잭션은 관계 없이, Active 이중화 stop 후 Standby 서버를 재시작 한 후 Standby에서 ALTER REPLICATION rep START WITH OFFLINE; 이라고 하면 에러가 날 것입니다. TC는 한번 확인해 보겠습니다. 
이 상황은 Active-Standby가 모두 비정상 종료 한 후 Standby에서 재시작후 ALTER REPLICATION rep START WITH OFFLINE; 을 할 수 없는 상황을 설명한 것입니다.

Reply:
이승원 팀장님이 추가하신 구동후라는 말이 Active 서버가 다음에 오는게 자연스러워 보입니다.
단, Active 서버가 **구동 후에** Standby 서버 쪽으로 한번이라도 이중화를 수행한 적이 있는 경우에만, Standby 서버에서 오프라인 옵션을 사용할 수 있다.

Reply:
제 의도는 그것이 아니라, Standby 서버가 비정상 종료했다가 다시 뜨면 Offline Replication의 Remote Meta가 없어서 Offline을 실패합니다. 그것에 대한 설명으로 Standby가 구동후에 한번이라도 이중화 Start가 성공한 후에만 Offline을 사용할 수 있다는 의미입니다.

Reply:
아래와 같은 상황임을 이승원 팀장님에게 확인 받았습니다. 

- 이중화 객체 생성만 한 상태에서 오프라인 이중화 시작한 경우 
iSQL> alter replication rep start with offline;
[ERR-610C2 : The meta information for the offline sender cannot be found in memory.]

- 오프라인 이중화를 설정한 지역 서버에서 Altibase 서버 재시작
iSQL> alter replication rep start with offline;
[ERR-610C2 : The meta information for the offline sender cannot be found in memory.]
$ altierr 0x610C2
0x610C2 ( 397506) rpERR_ABORT_RP_NOT_FOUND_OFFMETA The meta information for the offline sender cannot be found in memory. 
 *Cause:
 The remote sender has not been started since the local server was started.
 *Action:
 Start the replication in the remote server.


위 에러 메시지에서 The meta information 는 .. 송신자가 한 번이라도 이중화 메타 테이블 정로를 전송한 적이 있어야 한다는 의미로 이해하였고, 이승원 팀장님에게 아래와 같은 답변을 받았습니다. 

> 네 그 의미와 같습니다. sender가 메타 정보를 전송하면 그것을 receiver(standby)쪽에서 이중화 관리자가 메모리에 저장합니다. 그리고, 메모리에만 유지해서 서버가 내려가면 사라집니다. 이후에 다시 sender가 접속하면 그 때 받은 정보를 다시 메모리에 저장합니다.